httpbakery/checkers.go

```diff
@@ -4,38 +4,66 @@ import (
 	"net"
 	"net/http"
 
+	"golang.org/x/net/context"
 	"gopkg.in/errgo.v1"
 
 	"gopkg.in/macaroon-bakery.v2-unstable/bakery/checkers"
 )
 
-type httpContext struct {
-	req *http.Request
+type httpRequestKey struct{}
+
+func ContextWithRequest(ctxt context.Context, req *http.Request) context.Context {
```

Comment:
I'm not certain adding the whole request to the context is a good idea. How about something like?
```go
func ContextFromRequest(parentt context.Context, req *http.Request) context.Context {
	ctx := parent
	if ctx == nil {
		ctx = req.Context() // or context.Background() depending on go version
	}
	ctx = context.WithValue(ctx, ipAddrKey, req.RemoteAddr)
	ctx = context.WithValue(ctx, originKey, req.Header.Get("Origin"))
	return ctx
}